person_object.rb

```diff
@@ -1,11 +1,47 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    return @nameable.correct_name[0..9] if @nameable.correct_name.length > 10
+
+    @nameable.correct_name
+  end
+end
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @age = age
     @name = name
     @parent_permission = parent_permission
+    super
+  end
+
+  def correct_name
+    @name
```

Comment:
- [ OPTIONAL ] Great working on the functionalty so far but I highly recommend to make each class in seprate file to make your project more organized and clean and readable 💯✨✅🔥🔥🔥

Reply:
Noted. Thanks.